### ISSUE TITLE
Type definitions not working when tsconfig.json `module` is anything other than `CommonJS`

### ISSUE THREAD

**Issue by bdr99** [htunnicliff/jmap-jam#11]
Hello, and thanks for creating this library!

I have a TypeScript project that uses ES modules (package.json has `"type": "module"` and tsconfig.json has `"module": "node16"`). When I try to import jmap-jam in this project, it seems the type of `JamClient` is inferred as `any`:

![image](https://github.com/htunnicliff/jmap-jam/assets/2292715/e3ccb5b2-0e04-46cf-af2c-0d06dbda6af0)

If I change tsconfig.json to `"module": "commonjs"`, the type definitions work as expected:

![image](https://github.com/htunnicliff/jmap-jam/assets/2292715/711f340d-8c5a-42c1-90c6-3a2568756a92)

Unfortunately, this issue makes it difficult to use the library since TypeScript is not aware of any of the types. It is easy to reproduce this error with the following steps:

1. `mkdir test-project`
2. `cd test-project`
3. `npm init` (and accept defaults)
4. `npm install --save-dev typescript`
5. Open the project in VS Code
6. Add `"type": "module"` to package.json
7. Change `module` to `"node16"` in tsconfig.json
8. Create a source file index.ts and attempt to import `JamClient`

Any assistance would be appreciated!

Node.js version: 20.11.1
TypeScript version: 5.4.5
VS Code version: 1.88.1

**Comment by htunnicliff** [comment]
@bdr99 thank you for discovering this and providing a reproduction steps. I was able to reproduce this on my end as well.

I did some digging and realized that while I publish this package as ESM, the tsconfig.json was using Node's traditional module resolution.

I updated the module resolution to use Node16 (for ESM) and added file extensions to all imports and exports within the library.

I believe this is fixed in [v0.6.0](https://github.com/htunnicliff/jmap-jam/releases/tag/v0.6.0) – will you give it a try and let me know if it is working?
